solr/core/src/java/org/apache/solr/update/DocumentBuilder.java

```diff
@@ -151,63 +153,49 @@ public static Document toDocument(SolrInputDocument doc, IndexSchema schema, boo
 
       String name = field.getName();
       SchemaField sfield = schema.getFieldOrNull(name);
-      boolean used = false;
       
+      List<CopyField> copyFields = schema.getCopyFieldsList(name);
+      if( copyFields.size() == 0 ) copyFields = null;
+
       // Make sure it has the correct number
-      if( sfield!=null && !sfield.multiValued() && field.getValueCount() > 1 ) {
+      if( sfield!=null && !(sfield.getType() instanceof DenseVectorField) && !sfield.multiValued() && field.getValueCount() > 1 ) {
```

Comment:
subjective: add new specialised clauses at the end with short-circuit evaluation in mind
```suggestion
      if( sfield!=null && !sfield.multiValued() && field.getValueCount() > 1 && !(sfield.getType() instanceof DenseVectorField) ) {
```